### PR TITLE
Added main. Added JDBC DAO for Products

### PR DESCRIPTION
**Pull request by Nokafan**

### CODE REVIEW

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            }
+            return product;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Product " + product.getName() + " not created", e);
+        }
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        String query = "SELECT * from products where product_id = ?;";
```
Comment (OleksiiTromsa) [inline src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: Maybe let's add check for product_deleted here?

**Review by OleksiiTromsa — COMMENTED** [review]
Good job!

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 1 above)
Comment (Nokafan) [reply src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: Added

File: src/main/java/com/internet/shop/Main.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package com.internet.shop;
+
+import com.internet.shop.dao.jdbc.ProductDaoJdbcImpl;
+import com.internet.shop.model.Product;
+
+public class Main {
+
+    public static void main(String[] args) {
+        ProductDaoJdbcImpl productDaoJdbc = new ProductDaoJdbcImpl();
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/Main.java]: can we use injector here?

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 1 above)
Comment (Sofasmile) [inline src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: let's get only existing products

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all products from DB.", e);
+        }
+        return products;
+    }
+
+    @Override
+    public Product update(Product product) {
+        String query = "UPDATE products SET product_name = ?, product_price = ? "
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java:73]: can we update deleted data?

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            return product;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update product id= " + product.getId()
+                    + " name= " + product.getName(), e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE products SET product_deleted = '1' WHERE (product_id = ?)";
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: let's use TRUE/FALSE instead of 1/0

Comment (Sofasmile) [inline src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]:
```suggestion
        String query = "UPDATE products SET product_deleted = '1' WHERE product_id = ?";
```

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 1 above)
Comment (Sofasmile) [inline src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]:
```suggestion
        String query = "SELECT * FROM products WHERE product_id = ?;";
```

File: src/main/resources/init_db.sql [hunk 5]
```diff
@@ -0,0 +1,9 @@
+CREATE SCHEMA `internet_shop` DEFAULT CHARACTER SET utf8 ;
+
+CREATE TABLE `products` (
+                            `product_id` bigint NOT NULL AUTO_INCREMENT,
+                            `product_name` varchar(255) NOT NULL,
+                            `product_price` decimal(13,2) NOT NULL,
+                            `product_deleted` tinyint(1) NOT NULL DEFAULT '0',
+                            PRIMARY KEY (`product_id`)
+) ENGINE=InnoDB AUTO_INCREMENT=32 DEFAULT CHARSET=utf8;
```
Comment (Sofasmile) [inline src/main/resources/init_db.sql]: do we need to use AUTO_INCREMENT=32?

File: src/main/java/com/internet/shop/Main.java (hunk 2 above)
Comment (Nokafan) [reply src/main/java/com/internet/shop/Main.java]: Yes we can, already fixed :)

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 1 above)
Comment (Nokafan) [reply src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: Fixed.

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 3 above)
Comment (Nokafan) [reply src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java:73]: its all was fixed before ((( it wasn't pushed (((

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 4 above)
Comment (Nokafan) [reply src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: Fixed

Comment (Nokafan) [reply src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: Thx, fixed.

File: src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 1 above)
Comment (Nokafan) [reply src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: Fixed

File: src/main/resources/init_db.sql (hunk 5 above)
Comment (Nokafan) [reply src/main/resources/init_db.sql]: It is generated by workbench don't have any idea how AUTO_INCREMENT works perfect, increase by 1 each new element Removed everything after ')'

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**